### PR TITLE
Save draft, make route public, make route a draft and delete it.

### PR DESCRIPTION
**Pull request by skeie**

### CODE REVIEW

File: src/redux/reducers/userReducer.js [hunk 1]
```diff
@@ -24,7 +25,10 @@ const UserAuth = Record({
 
     // could have had a list here, one for each profile.
     // but probabz not necessary.
-    userProfile: null
+    userProfile: null,
+
+    //update route
+    updateRoute: false
```
Comment (sslash) [inline src/redux/reducers/userReducer.js]: what does updateRoute mean?

File: src/redux/reducers/userReducer.js [hunk 2]
```diff
@@ -134,7 +138,47 @@ export default function reducer (state = initialState, action = {}) {
...
 }
+
+/**
+* @param (array) routes, list of routes
+* @param (string) routeId, the id of the route to change
+* @param (string) property, can be either isPrivate, isDraft or isDeleted
+* @param (bool) value, can be either true or false
+*/
+
+function findRoute (routes, routeId, property, value) {
```
Comment (sslash) [inline src/redux/reducers/userReducer.js]: the function says it finds a route, but here it seems like it updates the route

File: src/redux/reducers/createRouteReducer.js [hunk 3]
```diff
@@ -250,6 +253,21 @@ export default function reducer (state = initialState, action = {}) {
         case types.ROUTE_FINISH_SUCCESS:
             return state.set('finito', true);
 
+        case types.ROUTES_CREATE_TOGGLE_DRAFT:
```
Comment (sslash) [inline src/redux/reducers/createRouteReducer.js]: IMO the way we update isPrivate / isActive / isDraft should be exactly the same, i.e same action names, same actionType name-structure. Also cant we just have a 
{
   updatingRoute: {
      error: 
      isUpdating:
 }
}

or something in the routes reducer, that is used for all three states. It's basicly just an update on a property

File: src/redux/constants/actionTypes.js [hunk 4]
```diff
@@ -233,4 +233,20 @@ export const BOOKMARKS_DELETE = 'BOOKMARKS_DELETE';
 export const BOOKMARKS_DELETE_SUCCESS = 'BOOKMARKS_DELETE_SUCCESS';
 export const BOOKMARKS_DELETE_FAIL = 'BOOKMARKS_DELETE_FAIL';
 
+export const ROUTES_CREATE_TOGGLE_DRAFT = 'ROUTES_CREATE_TOGGLE_DRAFT';
```
Comment (sslash) [inline src/redux/constants/actionTypes.js]: I would just 3 action types: update, update success, update fail

File: src/redux/actions/userActions.js [hunk 5]
```diff
@@ -111,3 +111,39 @@ export function bootstrapUser (user) {
         user
     };
 }
+
+export function deleteRoute (routeId) {
```
Comment (sslash) [inline src/redux/actions/userActions.js]: I would just have 1 action: updateUserRoute

File: src/models/Route.js [hunk 6]
```diff
@@ -24,7 +24,9 @@ const RouteRecord = Record({
     upvoteCount: 0,
     city: '',
     cityLocation: '',
-    isPrivate: false
+    isPrivate: false,
```
Comment (sslash) [inline src/models/Route.js:28]: nice!

File: src/models/Route.js [hunk 7]
```diff
@@ -24,7 +24,9 @@ const RouteRecord = Record({
...
+    isDraft: false,
+    isDeleted: false //used when a user deletes the route, and before the user refresh the site
```
Comment (sslash) [inline src/models/Route.js]: I would use the word active instead

File: src/redux/reducers/userReducer.js (hunk 2 above)
Comment (skeie) [inline src/redux/reducers/userReducer.js]: gp, forgott to rename it.

File: src/redux/reducers/userReducer.js (hunk 1 above)
Comment (skeie) [inline src/redux/reducers/userReducer.js]: If the user update's a route. Maybe call it something else.

File: src/redux/reducers/createRouteReducer.js (hunk 3 above)
Comment (skeie) [inline src/redux/reducers/createRouteReducer.js]: Yeah, i think we should have it the user reducer.

File: src/redux/actions/userActions.js (hunk 5 above)
Comment (skeie) [inline src/redux/actions/userActions.js]: And then pass in the URL?

File: src/models/Route.js (hunk 7 above)
Comment (skeie) [inline src/models/Route.js]: yeah

File: src/redux/actions/userActions.js (hunk 5 above)
Comment (sslash) [inline src/redux/actions/userActions.js]: Could the api url be the same too? just use the put route/id in routeRouter

Comment (skeie) [inline src/redux/actions/userActions.js]: I have changed all the three PR's to use updateroute :-). Have a look when you have the time.

File: src/redux/reducers/userReducer.js [hunk 8]
```diff
@@ -42,143 +37,132 @@ export default function reducer (state = initialState, action = {}) {
...
 * @param (array) routes, list of routes
-* @param (string) routeId, the id of the route to change
-* @param (string) property, can be either isPrivate, isDraft or isDeleted
-* @param (bool) value, can be either true or false
+* @param (object) updatedRoute, contains the route id and some properties. An exmaple of the object
+* can be {id: 1337, isActive: false, isDraft: true}
 */
 
-function findRoute (routes, routeId, property, value) {
+function updateRoute (routes, updatedRoute) {
```
Comment (sslash) [inline src/redux/reducers/userReducer.js:151]: udateRoutes (plural)

File: src/redux/reducers/__tests__/userReducer-test.js [hunk 9]
```diff
@@ -33,51 +33,29 @@ describe('user reducer', () => {
...
+        expect(afterState.get('routes')[0].get('isDraft')).to.equal(true);
+        expect(afterState.get('routes')[0].get('active')).to.equal(true);
 
-        const afterState = reducer(beforeState, {
-            type: types.USER_DELETE_ROUTE_SUCCESS,
-            meta: {routeId: 1337}
-        });
-        expect(afterState.get('routes')[0].get('isDeleted')).to.equal(true);
     });
-
```
Comment (sslash) [inline src/redux/reducers/__tests__/userReducer-test.js]: hadde gjort en liten test på at routes[1] ikke er blitt touchet :)

File: src/redux/constants/actionTypes.js [hunk 10]
```diff
@@ -233,10 +233,6 @@ export const BOOKMARKS_DELETE = 'BOOKMARKS_DELETE';
 export const BOOKMARKS_DELETE_SUCCESS = 'BOOKMARKS_DELETE_SUCCESS';
 export const BOOKMARKS_DELETE_FAIL = 'BOOKMARKS_DELETE_FAIL';
 
-export const ROUTES_CREATE_TOGGLE_DRAFT = 'ROUTES_CREATE_TOGGLE_DRAFT';
-export const ROUTES_CREATE_TOGGLE_DRAFT_SUCCESS = 'ROUTES_CREATE_TOGGLE_DRAFT_SUCCESS';
-export const ROUTES_CREATE_TOGGLE_DRAFT_FAIL = 'ROUTES_CREATE_TOGGLE_DRAFT_FAIL';
-
 export const USER_ROUTE_MAKE_PUBLIC = 'USER_ROUTE_MAKE_PUBLIC';
```
Comment (sslash) [inline src/redux/constants/actionTypes.js]: fjerne make public og delete også?

File: src/redux/actions/userActions.js [hunk 11]
```diff
@@ -112,38 +112,14 @@ export function bootstrapUser (user) {
...
-    return {
-        types: [
-            types.USER_ROUTE_MAKE_PUBLIC,
-            types.USER_ROUTE_MAKE_PUBLIC_SUCCESS,
-            types.USER_ROUTE_MAKE_PUBLIC_FAIL
-        ],
-        promise: ({req}) => req.post(`/routes/${routeId}/make-public`),
-        meta: { routeId }
-    }
+        promise: ({req}) => req.put(`/routes/${routeId}`, data),
```
Comment (sslash) [inline src/redux/actions/userActions.js:122]: aw yeah

**Pull request merged**